tests/unit/Lookup/MultiplexingEntityLookupTest.php

```diff
@@ -0,0 +1,165 @@
+		);
+	}
+
+	public function testGivenNotExistingForeignEntityIdFromKnownRepository_getEntityReturnsNull() {
+		$fooLookup = new InMemoryEntityLookup();
+		$fooLookup->addEntity( new FakeEntityDocument( new ItemId( 'foo:Q1' ) ) );
+		$multiplexingLookup = $this->getMultiplexingEntityLookupWithForeignLookups( array( 'foo' => $fooLookup, ) );
+
+		$this->assertNull( $multiplexingLookup->getEntity( new ItemId( 'foo:Q3000' ) ) );
+	}
+
+	public function testLookupExceptionsAreNotCaughtInGetEntity() {
+		$lookup = $this->getMock( EntityLookup::class );
+		$lookup->expects( $this->any() )
+			->method( 'getEntity' )
+			->willThrowException( new EntityLookupException( new ItemId( 'Q321' ) ) );
+
+		$multiplexingLookup = new MultiplexingEntityLookup( array( '' => $lookup ) );
+
+		$this->setExpectedException( EntityLookupException::class );
+		$multiplexingLookup->getEntity( new ItemId( 'Q321' ) );
+	}
+
+	public function testGivenEntityIdFromUnknownRepository_getEntityThrowsException() {
+		$multiplexingLookup = new MultiplexingEntityLookup(
+			array( '' => $this->getMock( EntityLookup::class ), )
+		);
+
+		$this->setExpectedException( UnknownForeignRepositoryException::class );
+
+		$multiplexingLookup->getEntity( new ItemId( 'foo:Q1' ) );
+	}
+
+	public function testGivenExistingLocalEntityId_hasEntityReturnsTrue() {
+		$lookup = new InMemoryEntityLookup();
+		$lookup->addEntity( new FakeEntityDocument( new ItemId( 'Q1' ) ) );
+		$multiplexingLookup = new MultiplexingEntityLookup( array( '' => $lookup ) );
+		$this->assertTrue( $multiplexingLookup->hasEntity( new ItemId( 'Q1' ) ) );
+	}
+
+	public function testGivenNotExistingLocalEntityId_getEntityReturnsFalse() {
+		$lookup = new InMemoryEntityLookup();
+		$multiplexingLookup = new MultiplexingEntityLookup( array( '' => $lookup ) );
+		$this->assertFalse( $multiplexingLookup->hasEntity( new ItemId( 'Q1' ) ) );
+	}
+
+	public function testGivenForeignEntityIdFromKnownRepository_hasEntityReturnsTrue() {
+		$fooLookup = new InMemoryEntityLookup();
+		$fooLookup->addEntity( new FakeEntityDocument( new ItemId( 'foo:Q1' ) ) );
+		$multiplexingLookup = $this->getMultiplexingEntityLookupWithForeignLookups( array( 'foo' => $fooLookup, ) );
+
+		$this->assertTrue( $multiplexingLookup->hasEntity( new ItemId( 'foo:Q1' ) ) );
+	}
+
+	public function testGivenNotExistingForeignEntityIdFromKnownRepository_hasEntityReturnsFalse() {
+		$fooLookup = new InMemoryEntityLookup();
+		$fooLookup->addEntity( new FakeEntityDocument( new ItemId( 'foo:Q1' ) ) );
+		$multiplexingLookup = $this->getMultiplexingEntityLookupWithForeignLookups( array( 'foo' => $fooLookup, ) );
+
+		$this->assertFalse( $multiplexingLookup->hasEntity( new ItemId( 'foo:Q3000' ) ) );
+	}
+
+	public function testLookupExceptionsAreNotCaughtInHasEntity() {
+		$lookup = $this->getMock( EntityLookup::class );
+		$lookup->expects( $this->any() )
+			->method( 'hasEntity' )
+			->willThrowException( new EntityLookupException( new ItemId( 'Q321' ) ) );
```

Comment:
Happy to see you using `$this->any()`. What about the method part though - is that needed or can you use `$this->anything()`?